src/components.jl

```diff
@@ -145,6 +145,17 @@ function _remove_component!(
     return component
 end
 
+"""
+Check to see if a component exists.
+"""
+function has_component(
+    ::Type{T},
+    components::Components,
+    name::AbstractString,
+) where {T <: InfrastructureSystemsComponent}
+    !isnothing(get_component(T, components, name))
```

Comment:
I think this should never throw an exception. Here is a proposal:

```suggestion
        !isconcretetype(T) && return !isempty(get_components_by_name(T, components, name))
        !haskey(components.data, T) && return false
        return haskey(components.data[T], name)
```